src/configupdater/parser.py

```diff
@@ -468,7 +471,7 @@ def _read(self, fp: Iterable[str], fpname: str, into: Document):
             first_nonspace = self.NONSPACECRE.search(line)
             cur_indent_level = first_nonspace.start() if first_nonspace else 0
             if cursect is not None and optname and cur_indent_level > indent_level:
-                cursect[optname].append(value)
+                cast(List, cursect[optname]).append(value)
```

Comment:
Is this the place you mentioned the code calls append without verifying for `None`?

Reply:
Yes, at least it's no obvious check for `is not None`. Since options without `=`  lead to options with a `None` value there is no possibility of a continuation line. Since we check here for a continuation this branch can never be taken in case of a `None` option. So this is no bug but really hard to figure out even for a human, so mypy will never ever get this.

Reply:
Indeed. I would never guess that 😅